### PR TITLE
add tests for connect timeout handling

### PR DESCRIPTION
**Pull request by pkwarren**
Update connect_ext_test to add tests to verify error handling with
invalid timeouts.

**Before submitting your PR:** Please read through the contribution guide at https://github.com/bufbuild/connect-go/blob/main/.github/CONTRIBUTING.md

### CODE REVIEW

File: connect_ext_test.go [hunk 1]
```diff
@@ -575,6 +575,40 @@ func Test_CustomCompression(t *testing.T) {
 	assert.Equal(t, response.Msg, &pingv1.PingResponse{Text: request.Text})
 }
 
+func Test_ConnectTimeout(t *testing.T) {
```
Comment (mfridman) [inline connect_ext_test.go:578]: Just noticed, how come some test names are `Test_XYZ` and others `TestXYZ` .. maybe let's choose one for consistency? (without the underscore, unless that adds something?)

File: connect_ext_test.go [hunk 2]
```diff
@@ -575,6 +575,40 @@ func Test_CustomCompression(t *testing.T) {
...
+
+	t.Run("timeout_non_numeric", func(t *testing.T) {
+		t.Parallel()
+		assert.Equal(t, getPingResponseWithTimeout(t, "10s").StatusCode, http.StatusBadRequest)
+	})
+
+	t.Run("timeout_out_of_range", func(t *testing.T) {
+		t.Parallel()
+		assert.Equal(t, getPingResponseWithTimeout(t, "12345678901").StatusCode, http.StatusBadRequest)
+	})
```
[the remark this answers is not in the document]
Comment (mfridman) [reply connect_ext_test.go:609]: Do we already have a test case for the happy path, do we need one where the timeout is valid?

[the remark this answers is not in the document]
Comment (pkwarren) [reply connect_ext_test.go:609]: Yep - https://github.com/bufbuild/connect-go/blob/38baeb35be9870f922323ee612b8b0c66aa68223/connect_ext_test.go#L407-L430

File: connect_ext_test.go [hunk 3]
```diff
@@ -575,6 +575,40 @@ func Test_CustomCompression(t *testing.T) {
 	assert.Equal(t, response.Msg, &pingv1.PingResponse{Text: request.Text})
 }
 
+func TestConnectTimeout(t *testing.T) {
```
Comment (mfridman) [inline connect_ext_test.go:578]: Should this be called `TestInvalidHeaderTimeout` since we are not testing the timeout itself, but more malformed headers?

**Review by mfridman — APPROVED** [review]
lgtm. Minor nit on the naming, feel free to change to whatever makes sense:

https://github.com/bufbuild/connect-go/pull/281#discussion_r891726793

**Pull request merged**